src/agents/SovrynPoolScannerAgent.ts

```diff
@@ -0,0 +1,33 @@
+import {inject, injectable} from 'inversify';
+
+import {SovrynPoolRepository} from '../services/sovryn/SovrynPoolRepository.js';
+import {SovrynPoolScanner} from '../services/sovryn/SovrynPoolScanner.js';
+import {LoopAgent} from './LoopAgent.js';
+import {GraphClient} from '../services/graph/GraphClient.js';
+import Settings from '../types/Settings.js';
+
+@injectable()
+export class SovrynPoolScannerAgent extends LoopAgent {
+  backoffTime = 10000;
+
+  scanner: SovrynPoolScanner;
+
+  constructor(@inject('Settings') settings: Settings) {
+    super();
+    const poolReposity = new SovrynPoolRepository();
+    const graphClient = new GraphClient(settings.api.sovryn.apiKey);
```

Comment:
`settings` should be like for uniswap dex - multichain and multipool, the way you did it will work only for one instance. 

https://docs.google.com/document/d/1cBi7riaOJpid-5zv0HiV_-XmamgBp0GR34tIqCWZodA/edit#heading=h.7gtykmqy8uqa

> Solution should support many protocols on many blockchains. So eg. If there will be a copy/fork of Sovryn on the same blockchain and/or on another one, we should be able to integrate with it by just adding an additional setup.

Reply:
Sounds good, so do we want to eventually use this service on other blockchains? i.e. not only Rootstock.